lib/rules/at-rule-descriptor-no-unknown/index.cjs

```diff
@@ -0,0 +1,58 @@
+// NOTICE: This file is generated by Rollup. To modify it,
+// please instead edit the ESM counterpart and rebuild with Rollup (npm run build).
+'use strict';
+
+const cssTree = require('css-tree');
+const report = require('../../utils/report.cjs');
+const ruleMessages = require('../../utils/ruleMessages.cjs');
+const validateOptions = require('../../utils/validateOptions.cjs');
+
+const ruleName = 'at-rule-descriptor-no-unknown';
+
+const messages = ruleMessages(ruleName, {
+	rejected: (atRule, descriptor) =>
+		`Unexpected unknown descriptor "${descriptor}" for at-rule "${atRule}"`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/at-rule-descriptor-no-unknown',
+};
+
+/** @type {import('stylelint').CoreRules[ruleName]} */
+const rule = (primary) => {
+	return (root, result) => {
+		const validOptions = validateOptions(result, ruleName, { actual: primary });
+
+		if (!validOptions) {
+			return;
+		}
+
+		root.walkAtRules((atRule) => {
+			atRule.walkDecls((decl) => {
+				const { prop, value } = decl;
+
+				const { error } = cssTree.lexer.matchAtruleDescriptor(atRule.name, prop, value);
+
+				if (!error) return;
+
+				if (error.name !== 'SyntaxReferenceError') return;
+
```

Comment:
```suggestion

				if (!error.message.startsWith('Unknown at-rule descriptor')) return;
				
```

Currently, the rule also checks and flags unknown at-rules. Let's return early for anything that isn't an unknown descriptor error. This change should make the suggestion to the tests pass.

Reply:
Fixed in this commit.
[Skip unknown at-rules](https://github.com/stylelint/stylelint/pull/8197/commits/d44e66cbe99f7c57e663d196e4642c9cdc264ab3)